docs/development/internal_api.rst

```diff
@@ -12,3 +20,73 @@ Internal API
    :undoc-members:
    :private-members:
    :show-inheritance:
+
+Extensions
+----------
+.. automodule:: blocks.extensions.predicates
+   :undoc-members:
+   :members:
+   :private-members:
+   :show-inheritance:
+
+Monitoring
+----------
+.. automodule:: blocks.monitoring.aggregation
+   :undoc-members:
+   :members:
+   :private-members:
+   :show-inheritance:
+
+.. automodule:: blocks.monitoring.evaluators
+   :undoc-members:
+   :members:
+   :private-members:
+   :show-inheritance:
+
+Search
+------
+.. automodule:: blocks.search
```

Comment:
I should have corrected. Let me know if it is ok before merging please and I will rebase.